Disabled: migrate to TypeScript

torounit opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

Part of https://github.com/WordPress/gutenberg/issues/35744

## What?
Converts the `Disabled` component to TypeScript.

## Testing Instructions

1. `npm run storybook:dev`
2. check `Disabled`

## Screenshots or screencast <!-- if applicable -->

torounit committed:
refactor storybook

torounit committed:
convert to typescript

torounit committed:
fix ComponentProps

torounit committed:
add changelog.

walbo commented on packages/components/src/disabled/index.tsx:
```diff
@@ -13,24 +13,19 @@ import { createContext } from '@wordpress/element';
  */
 import { StyledWrapper } from './styles/disabled-styles';
+import type { DisabledProps } from './types';
+import type { RefCallback } from 'react';
```
Imports from react should be under thee `External dependencies` comment.

walbo commented on packages/components/src/disabled/stories/index.tsx:
```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta, ComponentStory } from '@storybook/react';
```
I belive is common to import external dependencies first.

```suggestion
/**
 * External dependencies
 */
import type { ComponentMeta, ComponentStory } from '@storybook/react';

/**
 * WordPress dependencies
 */
import { useState } from '@wordpress/element';

/**
 * Internal dependencies
 */
import Disabled from '../';
import Button from '../../button/';
import SelectControl from '../../select-control/';
import TextControl from '../../text-control/';
import TextareaControl from '../../textarea-control/';
```

walbo commented on packages/components/src/disabled/index.tsx:
```diff
@@ -13,24 +13,19 @@ import { createContext } from '@wordpress/element';
+	children,
+	isDisabled = true,
+	...props
+}: DisabledProps ) {
```
Should use `WordPressComponentProps` here instead of `ComponentProps` in `types.ts`

See https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#refactoring-a-component-to-typescript Point 7 - vi

walbo commented on packages/components/src/disabled/index.tsx:
```diff
@@ -13,24 +13,19 @@ import { createContext } from '@wordpress/element';
-function Disabled( { className, children, isDisabled = true, ...props } ) {
-	/** @type {import('react').RefCallback<HTMLDivElement>} */
-	const ref = useDisabled();
+function Disabled( {
```
Would be great to add a description and a usage example comment to the function.

walbo commented on packages/components/src/disabled/stories/index.tsx:
```diff
@@ -0,0 +1,85 @@
+	);
+};
+
+export const DisabledWithProp: ComponentStory< typeof Disabled > = () => {
```
I doesn't look like the controls for the story works as expected. If not sure if this story is necessary when controls is fixed, since you then can use the controls to control the `isDisabled` prop.

torounit replied:
This story is not needed. Deleted it.

walbo reviewed (commented):
Good job so far! Left a few comments.

Looks like you also need to rebase the PR to get the changelog entry to the newest `Unreleased` section.

torounit committed:
fix imports order

torounit committed:
use WordPressComponentProps.

torounit committed:
add example

walbo commented on packages/components/src/disabled/index.tsx:
````diff
@@ -0,0 +1,82 @@
+ * `Disabled` is a component which disables descendant tabbable elements and prevents pointer interaction.
+ *
+ * @example
+ * ```js
````
Lets specify the example as `jsx` code

```suggestion
 * ```jsx
```

ciampo commented on packages/components/src/disabled/index.tsx:
```diff
@@ -0,0 +1,82 @@
+	isDisabled = true,
+	...props
+}: WordPressComponentProps< DisabledProps, 'div' > ) {
+	const ref: RefCallback< HTMLElement > = useDisabled();
```
Since `useDisabled` (and the underlying `useRefEffect` hook) are typed, there shouldn't be a need to explicitly add a type to `ref` here:

```suggestion
	const ref = useDisabled();
```

unless we want to be specific about the `ref` targeting a `div` element (instead of a generic HTML element), similarly to how `ref` was typed before this PR:

```suggestion
	const ref: RefCallback< HTMLDivElement > = useDisabled();
```

ciampo commented on packages/components/src/disabled/index.tsx:
```diff
@@ -0,0 +1,82 @@
+			<StyledWrapper
+				ref={ ref }
+				className={ classnames( className, 'components-disabled' ) }
+				{ ...props }
```
Probably something that should not be addressed in this PR, as it's also on `trunk`, but I noticed that the remaining `...props` are only applied to the `StyledWrapper` when `isDisabled` is `true`. It would be great normalize the situation, probably the best way could be to always render a wrapper `div` getting the `...props` regardless of the value of `isDisabled` (making sure that it doesn't cause any breaking changes).

(cc in @mirka in case she has thoughts on this)

mirka replied:
Not a strong opinion, but I'm a bit inclined to leave it alone until we get an actual request for it. I was wondering what other prop (other than `className`) anyone would want to pass to something like this, when I found [this](https://github.com/WordPress/gutenberg/blob/6bc2f8eb51359382a7178591a430a3141a4394b7/packages/block-editor/src/components/block-preview/auto.js#L67-L69). I guess that means third-party consumers could be passing `style` props relying on the fact that it only applies when `isDisabled` is true 😅 So, until there is an actual need for this to be normalized, there might not be enough benefit to offset the breaking cost.

ciampo commented on packages/components/src/disabled/types.ts:
```diff
@@ -0,0 +1,18 @@
+/**
+ * External dependencies
+ */
+import type { ComponentProps } from 'react';
+
+export interface DisabledProps extends ComponentProps< 'div' > {
```
Since we're using ges/components/src/ui/context/wordpress-component.ts#L13-L24, there's no need to extend `ComponentProps` here

```suggestion
export interface DisabledProps {
```

ciampo reviewed (commented):
Thank you @torounit for working on this, and @walbo for the first round of review!

Is there a chance we could also update the unit tests to TypeScript ?

torounit committed:
Update packages/components/src/disabled/types.ts

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>

torounit committed:
Update packages/components/src/disabled/index.tsx

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>

walbo commented on packages/components/src/disabled/index.tsx:
```diff
@@ -0,0 +1,82 @@
+/**
+ * `Disabled` is a component which disables descendant tabbable elements and prevents pointer interaction.
+ *
+ * @example
```
The `@example` JSDoc keyword is not recognized by StoryBook's docgen, so it should be removed.
 
 See https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#refactoring-a-component-to-typescript point 8.

```suggestion
```

ciampo commented on packages/components/src/disabled/stories/index.tsx:
```diff
@@ -0,0 +1,67 @@
+const meta: ComponentMeta< typeof Disabled > = {
+	title: 'Components/Disabled',
+	component: Disabled,
+	parameters: {
```
Let's also disable the interactive controls for the `as` prop, as we normally do for other components.

torounit committed:
Update packages/components/src/disabled/index.tsx

Co-authored-by: Petter Walbø Johnsgård <petter@dekode.no>

torounit committed:
Update packages/components/src/disabled/index.tsx

Co-authored-by: Petter Walbø Johnsgård <petter@dekode.no>

torounit committed:
Remove unused imports.

torounit commented:
@ciampo 

Thanks for the review.

I will update Unit test to ts as well. Will ping you when done!

torounit committed:
Rewrite to Testing-library `will disable all fields`

torounit committed:
Rewrite to Testing-library `should cleanly un-disable via reconciliation`

torounit committed:
use rerender

torounit committed:
refactor test.

torounit committed:
replace react-dom/test-utils to testing-library.

torounit committed:
remove unnecessary MutationObserver stab.
@see https://github.com/WordPress/gutenberg/pull/20766 https://github.com/WordPress/gutenberg/pull/20514/

torounit committed:
Convert to typescript.

torounit committed:
add story for contentEditable

torounit committed:
add control settings.

torounit commented on packages/components/src/disabled/stories/index.tsx:
```diff
@@ -0,0 +1,82 @@
+	argTypes: {
+		as: { control: { type: null } },
+		children: { control: { type: null } },
+		ref: { control: { type: null } },
```
I'm thinking it would be better to remove the ref from Control, any ideas?

![スクリーンショット 2022-08-18 11 45 56](https://user-images.githubusercontent.com/1908815/185281840-d6973e0b-6d6f-489e-af37-5ce41aa81b20.png)

ciampo replied:
What we've been doing in similar situations is to explicitly `Omit` the `ref` prop from the component's props:

```diff
diff --git a/packages/components/src/disabled/index.tsx b/packages/components/src/disabled/index.tsx
index 9b8fdd230f..df65c12924 100644
--- a/packages/components/src/disabled/index.tsx
+++ b/packages/components/src/disabled/index.tsx
@@ -49,12 +49,17 @@ const { Consumer, Provider } = Context;
  * };
  * ```
  */
-function Disabled( {
-	className,
-	children,
-	isDisabled = true,
-	...props
-}: WordPressComponentProps< DisabledProps, 'div' > ) {
+function Disabled(
+	// ref is omitted until we have `WordPressComponentPropsWithoutRef` or add
+	// ref forwarding to Disabled.
+	props: Omit< WordPressComponentProps< DisabledProps, 'div' >, 'ref' >
+) {
+	const {
+		className,
+		children,
+		isDisabled = true,
+		...additionalProps
+	} = props;
 	const ref = useDisabled();
 
 	if ( ! isDisabled ) {
@@ -66,7 +71,7 @@ function Disabled( {
 			<StyledWrapper
 				ref={ ref }
 				className={ classnames( className, 'components-disabled' ) }
-				{ ...props }
+				{ ...additionalProps }
 			>
 				{ children }
 			</StyledWrapper>
diff --git a/packages/components/src/disabled/stories/index.tsx b/packages/components/src/disabled/stories/index.tsx
index 8af37f7066..c59f8e1412 100644
--- a/packages/components/src/disabled/stories/index.tsx
+++ b/packages/components/src/disabled/stories/index.tsx
@@ -22,7 +22,6 @@ const meta: ComponentMeta< typeof Disabled > = {
 	argTypes: {
 		as: { control: { type: null } },
 		children: { control: { type: null } },
-		ref: { control: { type: null } },
 	},
 	parameters: {
 		controls: {
```

torounit commented on packages/components/src/disabled/test/index.js:
```diff
@@ -1,240 +0,0 @@
-} );
-
-describe( 'Disabled', () => {
-	let MutationObserver;
-
-	beforeAll( () => {
-		MutationObserver = window.MutationObserver;
-		window.MutationObserver = function () {};
-		window.MutationObserver.prototype = {
-			observe() {},
-			disconnect() {},
-		};
-	} );
-
-	afterAll( () => {
-		window.MutationObserver = MutationObserver;
-	} );
```
This stub is no longer needed and was removed.

https://github.com/WordPress/gutenberg/pull/20766 
https://github.com/WordPress/gutenberg/pull/20514

torounit committed:
fix changelog

torounit commented:
@ciampo

Changed testing to TypeScript and testing-library.

torounit commented on packages/components/src/disabled/stories/index.tsx:
```diff
@@ -0,0 +1,82 @@
+	);
+};
+
+export const ContentEditable: ComponentStory< typeof Disabled > = ( args ) => {
+	return (
+		<Disabled { ...args }>
+			<div contentEditable tabIndex={ 0 }>
+				contentEditable
+			</div>
+		</Disabled>
+	);
+};
```
Added story for contentEditable elements.

ciampo commented on packages/components/CHANGELOG.md:
```diff
@@ -125,6 +126,7 @@
 -   `ComboboxControl`: Replace `keyboardEvent.keyCode` with `keyboardEvent.code`([#42569](https://github.com/WordPress/gutenberg/pull/42569)).
 -   `ComboboxControl`: Add support for uncontrolled mode ([#42752](https://github.com/WordPress/gutenberg/pull/42752)).
 
+-
```
```suggestion
```

ciampo commented on packages/components/src/disabled/test/index.tsx:
```diff
@@ -0,0 +1,153 @@
+		);
+
+		const input = container.querySelector( 'form input' );
+		const div = container.querySelector( 'form div' );
```
Let's avoid querying the DOM explicitly, and instead rely more on the semantics (which allows us to test our components from the point of view of the user)

<details>

<summary>Suggestion:</summary>

```diff
diff --git a/packages/components/src/disabled/test/index.tsx b/packages/components/src/disabled/test/index.tsx
index a4dcfb0f7d..e94be9cbe8 100644
--- a/packages/components/src/disabled/test/index.tsx
+++ b/packages/components/src/disabled/test/index.tsx
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 
 /**
  * Internal dependencies
@@ -40,19 +40,19 @@ describe( 'Disabled', () => {
 	const Form = () => (
 		<form>
 			<input />
-			<div contentEditable tabIndex={ 0 } />
+			<div title="edit my content" contentEditable tabIndex={ 0 } />
 		</form>
 	);
 
 	it( 'will disable all fields', () => {
-		const { container } = render(
+		render(
 			<Disabled>
 				<Form />
 			</Disabled>
 		);
 
-		const input = container.querySelector( 'form input' );
-		const div = container.querySelector( 'form div' );
+		const input = screen.getByRole( 'textbox' );
+		const div = screen.getByTitle( 'edit my content' );
 		expect( input ).toBeDisabled();
 		expect( div ).toHaveAttribute( 'contenteditable', 'false' );
 		expect( div ).not.toHaveAttribute( 'tabindex' );
@@ -69,11 +69,11 @@ describe( 'Disabled', () => {
 				<Form />
 			);
 
-		const { container, rerender } = render( <MaybeDisable /> );
+		const { rerender } = render( <MaybeDisable /> );
 		rerender( <MaybeDisable isDisabled={ false } /> );
 
-		const input = container.querySelector( 'form input' );
-		const div = container.querySelector( 'form div' );
+		const input = screen.getByRole( 'textbox' );
+		const div = screen.getByTitle( 'edit my content' );
 
 		expect( input ).not.toBeDisabled();
 		expect( div ).toHaveAttribute( 'contenteditable', 'true' );
@@ -87,10 +87,10 @@ describe( 'Disabled', () => {
 			</Disabled>
 		);
 
-		const { container, rerender } = render( <MaybeDisable /> );
+		const { rerender } = render( <MaybeDisable /> );
 
-		const input = container.querySelector( 'form input' );
-		const div = container.querySelector( 'form div' );
+		const input = screen.getByRole( 'textbox' );
+		const div = screen.getByTitle( 'edit my content' );
 
 		expect( input ).toBeDisabled();
 		expect( div ).toHaveAttribute( 'contenteditable', 'false' );
```
</details>

For more clarity, we may even rename the `div` variable to `contentEditable`?

ciampo commented on packages/components/src/disabled/test/index.tsx:
```diff
@@ -0,0 +1,153 @@
+			);
+
+		const { container, rerender } = render( <MaybeDisable /> );
+		rerender( <MaybeDisable isDisabled={ false } /> );
+
+		const input = container.querySelector( 'form input' );
+		const div = container.querySelector( 'form div' );
```
Let's add a few checks here to show that we expect the form elements to be disabled:

```suggestion
		const input = screen.getByRole( 'textbox' );
		const contentEditable = screen.getByTitle( 'edit my content' );
		
		expect( input ).toBeDisabled();
		expect( contentEditable ).toHaveAttribute( 'contenteditable', 'false' );

		rerender( <MaybeDisable isDisabled={ false } /> );
```

ciampo commented on packages/components/src/disabled/test/index.tsx:
```diff
@@ -0,0 +1,153 @@
+		expect( div ).toHaveAttribute( 'contenteditable', 'true' );
+	} );
+
+	// Ideally, we'd have two more test cases here:
+	//
+	//  - it( 'will disable all fields on component render change' )
+	//  - it( 'will disable all fields on sneaky DOM manipulation' )
+	//
+	// Alas, JSDOM does not support MutationObserver:
+	//
+	//  https://github.com/jsdom/jsdom/issues/639
```
We should look into adding these tests, since the `MutationObserver` is now supported — but that can definitely happen in a follow-up (if you have capacity to work on it!)

ciampo commented on packages/components/src/disabled/test/index.tsx:
```diff
@@ -0,0 +1,153 @@
+	//
+	//  https://github.com/jsdom/jsdom/issues/639
+
+	describe( 'Consumer', () => {
```
We could add a similar list of tests that consume the Content via `useContent` instead than using `Disabled.Consumer` — but that's definitely a task for a separate PR!

ciampo commented on packages/components/src/disabled/test/index.tsx:
```diff
@@ -0,0 +1,153 @@
+			);
+		}
+
+		test( "lets components know that they're disabled via context", () => {
+			const { container } = render(
+				<Disabled>
+					<DisabledStatus />
+				</Disabled>
+			);
+
+			const wrapperElement = container.querySelector( 'p' );
+			expect( wrapperElement ).toHaveTextContent( 'Disabled' );
+		} );
+
+		test( "lets components know that they're not disabled via context when isDisabled is false", () => {
+			const { container } = render(
+				<Disabled isDisabled={ false }>
+					<DisabledStatus />
+				</Disabled>
+			);
+			const wrapperElement = container.querySelector( 'p' );
+			expect( wrapperElement ).toHaveTextContent( 'Not disabled' );
+		} );
+
+		test( "lets components know that they're not disabled via context", () => {
+			const { container } = render( <DisabledStatus /> );
+			const wrapperElement = container.querySelector( 'p' );
+			expect( wrapperElement ).toHaveTextContent( 'Not disabled' );
+		} );
+	} );
```
Let's not test implementation details by querying the DOM directly — we can use `screen.getByText` instead:

<details>

<summary>Click to expand</summary>

```diff
diff --git a/packages/components/src/disabled/test/index.tsx b/packages/components/src/disabled/test/index.tsx
index a4dcfb0f7d..b3c4d3728a 100644
--- a/packages/components/src/disabled/test/index.tsx
+++ b/packages/components/src/disabled/test/index.tsx
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 
 /**
  * Internal dependencies
@@ -124,30 +124,29 @@ describe( 'Disabled', () => {
 		}
 
 		test( "lets components know that they're disabled via context", () => {
-			const { container } = render(
+			render(
 				<Disabled>
 					<DisabledStatus />
 				</Disabled>
 			);
 
-			const wrapperElement = container.querySelector( 'p' );
-			expect( wrapperElement ).toHaveTextContent( 'Disabled' );
+			expect( screen.getByText( 'Disabled' ) ).toBeInTheDocument();
 		} );
 
 		test( "lets components know that they're not disabled via context when isDisabled is false", () => {
-			const { container } = render(
+			render(
 				<Disabled isDisabled={ false }>
 					<DisabledStatus />
 				</Disabled>
 			);
-			const wrapperElement = container.querySelector( 'p' );
-			expect( wrapperElement ).toHaveTextContent( 'Not disabled' );
+
+			expect( screen.getByText( 'Not disabled' ) ).toBeInTheDocument();
 		} );
 
-		test( "lets components know that they're not disabled via context", () => {
-			const { container } = render( <DisabledStatus /> );
-			const wrapperElement = container.querySelector( 'p' );
-			expect( wrapperElement ).toHaveTextContent( 'Not disabled' );
+		test( "lets components know that they're not disabled via context when the Disabled component is not rendered at all", () => {
+			render( <DisabledStatus /> );
+
+			expect( screen.getByText( 'Not disabled' ) ).toBeInTheDocument();
 		} );
 	} );
 } );

```
</details>

ciampo commented on packages/components/src/disabled/test/index.tsx:
```diff
@@ -0,0 +1,153 @@
+			expect( wrapperElement ).toHaveTextContent( 'Not disabled' );
+		} );
+
+		test( "lets components know that they're not disabled via context", () => {
```
```suggestion
		test( "lets components know that they're not disabled via context when the Disabled component is not rendered at all", () => {
```

ciampo reviewed (commented):
Thank you @torounit  for addressing the feedback so far!

I've left a few minor comments, but we should be quite close to being able to merge this PR!

One thing that I noticed during my tests, is that changing `isDisabled` from `false` to `true` will completely erase the value of the form elements:

https://user-images.githubusercontent.com/1083581/185608164-655d22e4-b8ff-48d0-b2f2-7b021a07cdc2.mp4

This is **not** a blocker for this PR, since the same behavior [can be observed on `trunk` ](https://wordpress.github.io/gutenberg/?path=/story/components-disabled--disabled-with-prop) — but at the same time, I'm not sure if we caught a bug or if the component was always supposed to work like this (cc @mirka). @gziolo , I see you had participate into some of work on this component (https://github.com/WordPress/gutenberg/pull/20514 and https://github.com/WordPress/gutenberg/pull/20766), do you remember if this behavior has always been around ?

gziolo commented:
> I see you had participate into some of work on this component (https://github.com/WordPress/gutenberg/pull/20514 and https://github.com/WordPress/gutenberg/pull/20766), do you remember if this behavior has always been around ?

I don't recall that at all. If it isn't covered by unit tests, then you can make your best judgment 👍🏻 

The behavior you described seems to be a bug.

ciampo commented:
> > I see you had participate into some of work on this component (#20514 and #20766), do you remember if this behavior has always been around ?
> 
> I don't recall that at all. If it isn't covered by unit tests, then you can make your best judgment 👍🏻
> 
> The behavior you described seems to be a bug.

Alright, let's treat it as a bug — we'll try to trace when it was introduced and work on a fix + unit tests.

torounit committed:
Update packages/components/CHANGELOG.md

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>

torounit committed:
Omit ref.

torounit committed:
avoid querying

torounit committed:
rename div.

torounit committed:
test before rerender

torounit committed:
Simplify

torounit committed:
Update packages/components/src/disabled/test/index.tsx

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>

torounit committed:
Merge remote-tracking branch 'origin/refactoring/components/disabled' into refactoring/components/disabled

# Conflicts:
#	packages/components/src/disabled/test/index.tsx

torounit committed:
test for sneaky DOM manipulation

torounit committed:
Fix `isDisabled` so that it keeps its value even if it is changed.

torounit commented:
@ciampo https://github.com/WordPress/gutenberg/pull/42708/commits/28820f041afe8f1818c749209d54bcefd97bce57

fixed it so that the values are retained, what do you think?

torounit committed:
add default args

ciampo commented:
> @ciampo [28820f0](https://github.com/WordPress/gutenberg/commit/28820f041afe8f1818c749209d54bcefd97bce57)
> 
> fixed it so that the values are retained, what do you think?

The approach looks promising, but let's work on it in a separate PR to keep this one focused on the TypeScript migration. 

Would you be able to remove https://github.com/WordPress/gutenberg/commit/28820f041afe8f1818c749209d54bcefd97bce57 from this PR and open a new PR with that fix instead?

Once that's done and CHANGELOG conflicts are solved, we should be able to merge this one 🚀

ciampo commented on packages/components/src/disabled/test/index.tsx:
```diff
@@ -0,0 +1,178 @@
+		expect( contentEditable ).toHaveAttribute( 'contenteditable', 'true' );
+	} );
+
+	// Ideally, we'd have two more test cases here:
+	//
+	//  - it( 'will disable all fields on component render change' )
+	//
```
I'm not really sure that the original author meant with this suggestion — we can probably remove this comment

torounit replied:
I didn't really know what to write in this test.  🥴 Thanks!

torounit committed:
Revert "Fix `isDisabled` so that it keeps its value even if it is changed."

This reverts commit 28820f041afe8f1818c749209d54bcefd97bce57.

torounit committed:
Merge branch 'trunk' into refactoring/components/disabled

torounit commented:
@ciampo 

Removed https://github.com/WordPress/gutenberg/commit/28820f041afe8f1818c749209d54bcefd97bce57 and
CHANGELOG was updated.

Thanks!

ciampo commented on packages/components/src/disabled/test/index.tsx:
```diff
@@ -0,0 +1,178 @@
+		expect( contentEditable ).toHaveAttribute( 'contenteditable', 'true' );
+	} );
+
+	// Ideally, we'd have two more test cases here:
+	//
+	//  - it( 'will disable all fields on component render change' )
+	//
```
```suggestion
```

ciampo committed:
Update packages/components/src/disabled/test/index.tsx

ciampo commented on packages/components/CHANGELOG.md:
```diff
@@ -48,6 +49,7 @@
 ### Experimental
 -   `FormTokenField`: add `__experimentalAutoSelectFirstMatch` prop to auto select the first matching suggestion on typing ([#42527](https://github.com/WordPress/gutenberg/pull/42527/)).
 
+
```
```suggestion
```

ciampo committed:
Update packages/components/CHANGELOG.md

ciampo commented:
Thank you @torounit !

If you have capacity, it would be great if you could open a follow-up PR to work on the [fix](https://github.com/WordPress/gutenberg/commit/28820f041afe8f1818c749209d54bcefd97bce57) that you had previously proposed
